src/codegen.zig

```diff
@@ -1480,6 +1480,8 @@ fn Function(comptime arch: std.Target.Cpu.Arch) type {
         }
 
         fn genArg(self: *Self, inst: *ir.Inst.Arg) !MCValue {
+            defer self.arg_index += 1;
```

Comment:
Can I suggest a fix that generates less code?

```suggestion
            const arg_index = self.arg_index;
            self.arg_index += 1;
```

(and then use `arg_index` instead of `self.arg_index` below)

Otherwise we needlessly have code that runs at every return point of the function (including every `try`)

With stage2 codegen it's less important, since we send all control flow to the same function return block, but stage1 duplicates the defer expressions at every return point.

Reply:
I agree, this makes more sense.